### PR TITLE
boot: add more error messages

### PR DESCRIPTION
**Pull request by axel-h**
Print detailed error messages to make porting and debugging easier.

### CODE REVIEW

File: src/kernel/boot.c [hunk 1]
```diff
@@ -644,6 +644,9 @@ BOOT_CODE bool_t create_untypeds(cap_t root_cnode_cap,
                 start, ndks_boot.reserved[i].start
             });
             if (!create_untypeds_for_region(root_cnode_cap, true, reg, first_untyped_slot)) {
+                printf("ERROR: creation of untypteds for device region #%u at"
```
Comment (Indanz) [inline src/kernel/boot.c:647]: untypteds  -> untypeds

File: src/kernel/boot.c [hunk 2]
```diff
@@ -657,12 +660,18 @@ BOOT_CODE bool_t create_untypeds(cap_t root_cnode_cap,
         });
 
         if (!create_untypeds_for_region(root_cnode_cap, true, reg, first_untyped_slot)) {
+            printf("ERROR: creation of untypteds for top device region"
```
Comment (Indanz) [inline src/kernel/boot.c:663]: untypteds  -> untypeds

**Review by Indanz — APPROVED** [review]
Otherwise looks good to me.

File: src/kernel/boot.c (hunk 1 above)
Comment (axel-h) [reply src/kernel/boot.c]: thanks, fixed.

File: src/kernel/boot.c (hunk 2 above)
Comment (axel-h) [reply src/kernel/boot.c]: thanks, fixed.

**Review by lsf37 — APPROVED** [review]
👍

File: src/kernel/boot.c [hunk 3]
```diff
@@ -644,6 +644,9 @@ BOOT_CODE bool_t create_untypeds(cap_t root_cnode_cap,
                 start, ndks_boot.reserved[i].start
             });
             if (!create_untypeds_for_region(root_cnode_cap, true, reg, first_untyped_slot)) {
+                printf("ERROR: creation of untypteds for device region #%u at"
+                       " [%"SEL4_PRIx_word"..%"SEL4_PRIx_word"] failed\n",
+                       (unsigned int)i, reg->start, reg->end);
```
[the remark this answers is not in the document]
Comment (axel-h) [reply src/kernel/boot.c]: yes, also fixed already.

**Pull request merged**